framework/src/base/NonlinearEigenSystem.C

```diff
@@ -135,4 +145,14 @@ NonlinearEigenSystem::getNthConvergedEigenvalue(dof_id_type n)
   return _transient_sys.get_eigenpair(n);
 }
 
+void
+NonlinearEigenSystem::addKernelToExtraWarehouse(MooseSharedPointer<KernelBase> kernel, THREAD_ID tid)
```

Comment:
Yes, right, only `EigenProblem` uses  `ExtraWarehouse` now, but in the future, we possibly  create `XXXProblem` to add `XXX kernels`. It would be much better if we could have a generic name.